test/unittest/utils/RefCountedPointerTests.cpp

```diff
@@ -153,21 +169,61 @@ TEST_F(RefCountedPointerTests, refcountedpointer_inactive)
 
 TEST_F(RefCountedPointerTests, refcounterpointer_deactivate_waits_for_no_references)
 {
+    struct WaitForAllOwners : public EntityOwner::SyncPoint
+    {
+        WaitForAllOwners()
+            : num_notifications_(0)
+        {
+        }
+
+        void notify_and_wait() override
+        {
+            std::unique_lock<std::mutex> lock(mutex_);
+            ++num_notifications_;
+            if (num_notifications_ == n_owners)
+            {
+                cv_.notify_all();
+            }
+
+            cv_.wait(lock, [this]() -> bool
+                    {
+                        return num_notifications_ >= n_owners;
```

Comment:
When the threads will leave from this wait? I think it is missing a num_notifications_++ in `wait_for_all_notifications`.

Reply:
`num_notifications_` is increased above in line 182. The idea is that the threads wait for all of them to be at this same point, so all of them start at the same time. That is why the method name is `notify_and_wait`

`wait_for_all_notifications` is used by the main thread to wait for all threads to start

Reply:
sorry, it a '>=' comparison. In my head I see only '>'. :)